### PR TITLE
Rebase code base

### PR DESCRIPTION
**Pull request by RalphSu**

### CODE REVIEW

**Comment by haoch** [comment]
The code sync should skip `pom.xml`

File: eagle-core/eagle-alert-parent/eagle-alert/alert-metadata-parent/pom.xml [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 	<modelVersion>4.0.0</modelVersion>
 
 	<parent>
 		<groupId>org.apache.eagle</groupId>
-		<artifactId>eagle-alert</artifactId>
-		<version>0.5.0-incubating-SNAPSHOT</version>
-		<relativePath>../pom.xml</relativePath>
+		<artifactId>alert-parent</artifactId>
+		<version>0.0.1-SNAPSHOT</version>
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-metadata-parent/pom.xml]: :-1: fix such kinds of version changes.

File: eagle-core/eagle-alert-parent/eagle-alert/alert-service/pom.xml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+				</exclusion>
+			</exclusions>
+		</dependency>
+	</dependencies>
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.eclipse.jetty</groupId>
+				<artifactId>jetty-maven-plugin</artifactId>
+				<version>9.2.11.v20150529</version>
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-service/pom.xml]: Resolve hard-coded pom version.

File: eagle-core/eagle-alert-parent/eagle-alert/pom.xml [hunk 3]
```diff
@@ -61,12 +154,17 @@
...
+			<dependency>
+				<groupId>org.apache.curator</groupId>
+				<artifactId>curator-test</artifactId>
+				<version>${curator.test.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.apache.curator</groupId>
 				<artifactId>curator-recipes</artifactId>
-				<version>${curator.version}</version>
+				<version>${curator-framework.version}</version>
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/pom.xml]: Resolve duplicated dependencies version.

File: eagle-core/eagle-alert-parent/eagle-alert/pom.xml [hunk 4]
```diff
@@ -107,7 +205,7 @@
 			<dependency>
 				<groupId>org.slf4j</groupId>
 				<artifactId>slf4j-log4j12</artifactId>
-				<version>${slf4j-log4j12.version}</version>
+				<version>${slf4j-log4j12.veresion}</version>
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/pom.xml]: Bad word `slf4j-log4j12.veresion`

**Comment by haoch** [comment]
There may be some version problem, please clean up following apache eagle's naming standard.

File: eagle-core/eagle-alert-parent/eagle-alert/pom.xml [hunk 5]
```diff
@@ -277,13 +286,92 @@
 				<artifactId>metrics-elasticsearch-reporter</artifactId>
 				<version>${metrics-elasticsearch-reporter.version}</version>
 			</dependency>
+			<dependency>
+				<groupId>commons-cli</groupId>
+				<artifactId>commons-cli</artifactId>
+				<version>${common.cli.version}</version>
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/pom.xml]: Maybe a bad practice, referring to version properties in nested module.

File: pom.xml [hunk 6]
```diff
@@ -292,8 +293,17 @@
         <mockito.version>1.10.19</mockito.version>
         <joda.version>2.2</joda.version>
         <metrics-elasticsearch-reporter.version>2.2.0</metrics-elasticsearch-reporter.version>
+
+        <common.cli.version>1.3.1</common.cli.version>
+
+        <!-- dropwizard -->
+        <dropwizard.version>0.7.1</dropwizard.version>
```
Comment (haoch) [inline pom.xml]: Why include dropwizard into eagle?

File: eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/simple/publishments.json [hunk 7]
```diff
@@ -9,21 +9,23 @@
...
+	"serializer" : "org.apache.eagle.alert.engine.publisher.impl.StringEventSerializer"
 },
 {
   "type": "org.apache.eagle.alert.engine.publisher.impl.AlertKafkaPublisher",
   "name":"kafka-testAlertStream",
   "policyIds": ["perfmon_cpu_host_check"],
   "dedupIntervalMin": "PT1M",
   "properties":{
-    "kafka_broker":"sandbox.hortonworks.com:6667",
+    "kafka_broker":"eagle-c3-lvs01-3-7965.lvs01.dev.ebayc3.com:9092",
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/simple/publishments.json]: Sensitive information!

File: eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/simple/application-integration.conf [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	"host" : "localhost",
+	"port" : 8080
+  },
+  "coordinatorService": {
+  	"host": "localhost",
+  	"port": "8080",
+  	"context" : "/rest"
+  },
+  "kafkaProducer": {
+  	"bootstrapServers": "eagle-c3-lvs01-3-7965.lvs01.dev.ebayc3.com:9092"
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/simple/application-integration.conf]: Sensitive information

File: eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/publishments1.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
+[
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/publishments1.json]: Sensitive information

File: eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/e2e/application-e2e.conf [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+	"host" : "localhost",
+	"port" : 8080
+  },
+  "coordinatorService": {
+  	"host": "localhost",
+  	"port": "8080",
+  	"context" : "/rest"
+  },
+  "kafkaProducer": {
+  	"bootstrapServers": "eagle-c3-lvs01-3-7965.lvs01.dev.ebayc3.com:9092"
```
Comment (haoch) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/resources/e2e/application-e2e.conf]: Sensitive information

**Comment by haoch** [comment]
@RalphSu Please clean sensitive information.

File: pom.xml (hunk 6 above)
Comment (RalphSu) [inline pom.xml]: Alert -service is able to be packaged as dropwizard jar

Comment (haoch) [inline pom.xml]: It should be ok. As dropwizard is a more full stack framework than jersey. As already included, we we may directly use some other features of dropwizard as well.

**Comment by haoch** [comment]
LGTM

@yonzhang @qingwen220 please help review as well. If no more problem, we could merge the PR tomorrow, it will be a milestone where we start to develop completely independently.

**Comment by haoch** [comment]
@zombieJ we could start to refactor the service after merging this patch.

**Comment by haoch** [comment]
Merged into `develop` branch. As to following problems caused by the PR, will refactor on apache repo directly.

**Comment by haoch** [comment]
Please close it.

**Pull request closed without merging**